hbase_sql/src/main/java/com/splicemachine/hbase/SpliceCompactionUtils.java

```diff
@@ -69,6 +73,46 @@ public static boolean forcePurgeDeletes(Store store) throws IOException {
                 txn.commit();
         }
 
-        return false;
+        return t.getDefaultValue();
+    }
+
+    public static boolean forcePurgeDeletes(Store store) throws IOException {
+        return extract(store, new TableDescriptorExtractor<Boolean>() {
+            @Override
+            public Boolean get(TableDescriptor td) {
+                return td.purgeDeletedRows();
+            }
+            @Override
+            public Boolean getDefaultValue() {
+                return false;
+            }
+        });
+    }
+
+    private static Long minRetentionPeriod(Store store) throws IOException { ;
+        return extract(store, new TableDescriptorExtractor<Long>() {
+            @Override
+            public Long get(TableDescriptor td) {
+                return td.minRetainedPeriod();
+            }
+            @Override
+            public Long getDefaultValue() {
+                return 0L;
+            }
+        });
+    }
+
+    public static long getTxnLowWatermark(Store store) throws IOException {
+        long lowTxnWatermark = TransactionsWatcher.getLowWatermarkTransaction();
+        Long minRetentionPeriod = SpliceCompactionUtils.minRetentionPeriod(store);
+        if(minRetentionPeriod == null || minRetentionPeriod == 0L) {
+            return lowTxnWatermark;
+        }
+        long minRetentionTs = System.currentTimeMillis() - minRetentionPeriod * 1000;
```

Comment:
I don't have the whole picture: does this account for time zone differences or is that not necessary?

Reply:
I think `AS OF` feature is based on server-time, this follows the same logic.